src/main/java/com/sphy/hotelmanagementapplication/domain/Room.java

```diff
@@ -0,0 +1,125 @@
+package com.sphy.hotelmanagementapplication.domain;
+
+import java.io.Serializable;
+import java.util.HashSet;
+import java.util.Set;
+
+import javax.persistence.Column;
+import javax.persistence.DiscriminatorValue;
+import javax.persistence.Entity;
+import javax.persistence.GeneratedValue;
+import javax.persistence.GenerationType;
+import javax.persistence.Id;
+import javax.persistence.JoinColumn;
+import javax.persistence.ManyToOne;
+import javax.persistence.OneToMany;
+
+@Entity(name = "rooms")
+@DiscriminatorValue("rooms")
+public class Room implements Serializable {
+    @Id
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
+    private Long Id;
+    @Column(name = "name")
+    private String name;
+    @Column(name = "luxurity")
+    private int luxurity;
+
+    @ManyToOne
+    @JoinColumn(name = "hotel_id")
+    private Hotel hotel;
+
+    @OneToMany(mappedBy = "rooms")
+    private Set<Order> orders =new HashSet<>();
+
+    private long price;
+
+
+
+    public Room() {
+    }
+
+    public Room(String name, int luxurity, long price) {
+        this.name = name;
+        this.luxurity = luxurity;
+
+        this.price = price;
+    }
+
+
+    public String getName() {
+        return name;
+    }
+
+    public void setName(String name) {
+        this.name = name;
+    }
+
+    public Hotel getHotel() {
+        return hotel;
+    }
+
+    public void setHotel(Hotel hotel) {
+        this.hotel = hotel;
+    }
+
+    public int getLuxurity() {
+        return luxurity;
+    }
+
+    public void setLuxurity(int luxurity) {
+        this.luxurity = luxurity;
+    }
+
+
+    public Long getId() {
```

Comment:
Either remove, or add super.getId(). The same thing must be done blow in the setter. It will produce an error otherwise

Reply:
Please check inside that method, i am not talking about its inheritance